lib/src/fnc/string.rs

```diff
@@ -75,6 +78,25 @@ pub fn replace((val, old, new): (String, String, String)) -> Result<Value, Error
 	Ok(val.replace(&old, &new).into())
 }
 
+fn do_replace_all(regex: &Regex, val: String, replacement: String) -> Result<Value, Error> {
+	Ok(regex.replace_all(&val, replacement).into_owned().into())
+}
+
+pub async fn replace_all(
+	ctx: &Context<'_>,
+	(val, regex, replacement): (String, String, String),
```

Comment:
I'm curious why `regex: String` instead of `regex: crate::sql::Regex`, which seems like the appropriate type. Note:
- I'm not sure how well coercion to `crate::sql::Regex` works (lack of support might block this idea)
- it might be difficult to run `async` caching code wherever `String`'s may be coerced to `crate::sql::Regex`

Reply:
One important part of this PR is that we are caching the Regex to avoid recompiling it. So, my approach was to use to have a cache using an Hashmap<String, Regex>. This would be efficient only if the same regexp is used several times in the same execution.
That said, I can assume that the Regexp compilation is optimized at the SQL parsing level (which can be done in another PR).
Ok, I am investigating this approach.

Reply:
Regarding the naming, I was indeed hesitating regarding names and packages, wondering if I should introduce a package for just one function:` regex::replace_all`
Thoughts?

Reply:
I finally have patched the existing string::replace function so it can take either a string or a regular expression. It covers also this request #1287